### PR TITLE
Adding support for TFS collections on Visual Studio.

### LINKED ISSUE #173: Support for collections on VisualStudio.com
GitLink doesn't recognize url's on Visual Studio which include a TFS project collection

### Steps to reproduce

1. Create an account on VSTS (VisualStudio.com)
2. Create a new project
3. Clone the project from its url, but add 'DefaultCollection' in the middle (e.g. https://youraccount.visualstudio.com/**DefaultCollection**/MyFirstProject/_git/MyFirstProject)

### Expected behaviour

GitLink will add a srcsrv section to the pdb using the VSTS provider (using tf.exe)

### Actual behaviour

The custom url provider is used instead of the VSTS provider. When debugging, the source isn't downloaded

### PR DESCRIPTION
**Pull request by ShaiNahum**
Fixes #173 .

### Checklist

- [x] I have included examples or tests
- [ ] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file
- [ ] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

- Adding support for TFS collections on VSTS (www.VisualStudio.com)

### CODE REVIEW

File: src/GitLink.Tests/Providers/VisualStudioTeamServicesProviderFacts.cs [hunk 1]
```diff
@@ -72,6 +72,15 @@ public void ReturnsValidProject2()
                 }
 
                 [TestCase]
+                public void ReturnsValidProject3()
```
Comment (GeertvanHorrik) [inline src/GitLink.Tests/Providers/VisualStudioTeamServicesProviderFacts.cs]: I think it's better to just refactor ReturnsValidProject into this:

```
[TestCase("https://CatenaLogic.visualstudio.com/_git/main-repo", "main-repo")]
[TestCase("https://CatenaLogic.visualstudio.com/BigProject/_git/main-repo", "BigProject")]
[TestCase("https://CatenaLogic.visualstudio.com/DefaultCollection/BigProject/_git/main-repo", "BigProject")]
public void ReturnsValidProject(string url, string expectedProjectName)
{
	var provider = new VisualStudioTeamServicesProvider();
	provider.Initialize(url);

	Assert.AreEqual(expectedProjectName, provider.ProjectName);
}
```

File: src/GitLink/Providers/VisualStudioTeamServicesProvider.cs [hunk 2]
```diff
@@ -13,7 +13,24 @@ namespace GitLink.Providers
 
     public class VisualStudioTeamServicesProvider : ProviderBase
     {
-        private static readonly Regex HostingUrlPattern = new Regex(@"(?<url>(?<companyurl>(?:https://)?(?<accountname>([a-zA-Z0-9\-\.]*)?)\.visualstudio\.com/)(?<project>[a-zA-Z0-9\-\.]*)/?_git//?(?<repo>[^/]+))");
+        // Matches the git origin URL, providing named capture groups
+        // Example match: https://user.visualstudio.com/DefaultCollection/MyFirstProject/_git/MyFirstRepo
+        private static readonly Regex HostingUrlPattern =
```
Comment (GeertvanHorrik) [inline src/GitLink/Providers/VisualStudioTeamServicesProvider.cs]: Nice, never saw regular expressions split up so nicely, well done.

**Comment by GeertvanHorrik** [comment]
Thanks a lot for your efforts! There is just 1 thing I'd like to see different, care for changing that?

**Comment by ShaiNahum** [comment]
Good call on the requested changes. I'm still not used to NUnit.
I've modified the tests according to your example

**Comment by GeertvanHorrik** [comment]
Sorry for the late review, missed notifications. All looks good now, thanks!

**Pull request merged**